common/src/main/java/io/confluent/connect/storage/common/GenericClassRecommender.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.storage.common;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+public class GenericClassRecommender implements ConfigDef.Recommender {
```

Comment:
I'm curious: why is this named GenericClassRecommender when it can hold any type of value?

Reply:
Of course, you're right. I thought I had mentioned but I didn't. Well, I didn't want to encourage a very broad usage because the pattern is not very solid (still, seems to be the best we can do right now without changing `ConfigDef` and `AbstractConfig`). As I mentioned, I started by targeting a generic implementation only to realize that some interfaces were not on the right modules. 

I'll change the name to reflect what you are pointing out, and hopefully it won't be used for more properties than absolutely necessary. Thanks